src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -619,4 +619,17 @@ public function getReturnStatementsAst()
 
         return $visitor->getReturnNodes();
     }
+
+    private function findTypeFromAst($namespace, $locatedSource, $type)
+    {
+        $objectType = (new FindTypeFromAst())->__invoke(
+            $this->reflector->getContextFactory()->createForNamespace(
```

Comment:
I think the big red light here, for me, is that in order to inject the ContextFactory (or similar) into the `ReflectionClass`, we need to pass it to all of the source locators, and all new `AstLocator` instances and then finally to the `NodeToReflection` class, when it is only required for the creation of the ReflectionClass.

Is there any reason, in principle, that it cannot be refactored to instantiate the `ReflectionClass` in the `Reflector`?